### PR TITLE
Package ppx_deriving_decoders.0.1

### PR DESCRIPTION
**Pull request by benbellick**
### `ppx_deriving_decoders.0.1`
Deriving Decoders using PPX
Using mattjbray/ocaml-decoders, use a ppx to automatically generate instances of a decoder for a particular type using PPX.

---
* Homepage: https://github.com/benbellick/ppx_deriving_decoders
* Source repo: git+https://github.com/benbellick/ppx_deriving_decoders.git
* Bug tracker: https://github.com/benbellick/ppx_deriving_decoders/issues

---
:camel: Pull-request generated by opam-publish v2.4.0

### CODE REVIEW

**Comment by shonfeder** [comment]
Hello :wave: Thank you for publishing your package!

Since it looks like this is your first time publishing a package to opam, allow me offer a warm welcome and a bit of an intro :)

We run a CI system to ensure that packages can be installed on all intended systems. You can see the results for your PR at https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/88cc1e996b0437ff2d34fd37af31645a80408a79 (or by clicking the Details link next to the opam-ci status check).

This shows the results of the builds and tests across our [build matrix](https://github.com/ocurrent/opam-repo-ci/blob/master/doc/platforms.md). Because we build and test on such a large matrix, it is very common for errors to be identified during package publication.

In my following comments and suggestions, I'll try to provide pointers and suggestions to get everything squared away. Please let us know if you have any questions, as we are here to help :)

**Review by shonfeder — COMMENTED** [review]
Aside from the lower bound error noted below, I see we are also seeing a failure to build decoders on arm32 (https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/88cc1e996b0437ff2d34fd37af31645a80408a79/variant/extras,arm32-ocaml-5.2,ppx_deriving_decoders.0.1) and x86_32 . In both cases it looks like the following:


```
#=== ERROR while compiling decoders.1.0.0 =====================================#
# context              2.4.0~alpha1~dev | linux/arm32 | ocaml-base-compiler.5.2.0 | file:///home/opam/opam-repository
# path                 ~/.opam/5.2/.opam-switch/build/decoders.1.0.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p decoders -j 15 @install
# exit-code            1
# env-file             ~/.opam/log/decoders-7-b537cc.env
# output-file          ~/.opam/log/decoders-7-b537cc.out
### output ###
# File "src/dune", lines 6-11, characters 0-105:
#  6 | (rule
#  7 |  (targets shims_let_ops_.ml)
#  8 |  (action
#  9 |   (with-stdout-to
# 10 |    %{targets}
# 11 |    (run ./gen/mkshims.exe))))
# Error: No rule found for src/gen/mkshims.exe
```

This is a problem with the docoder dependency rather than this package. I've reported it upstream in https://github.com/mattjbray/ocaml-decoders/issues/75

File: packages/ppx_deriving_decoders/ppx_deriving_decoders.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+authors: ["Ben Bellick <benbellick@pm.me>"]
+license: "MIT"
+tags: ["decoders" "decoding" "json" "ppx"]
+homepage: "https://github.com/benbellick/ppx_deriving_decoders"
+doc: "https://github.com/benbellick/ppx_deriving_decoders"
+bug-reports: "https://github.com/benbellick/ppx_deriving_decoders/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "3.11"}
+  "ppxlib"
```
[the remark this answers is not in the document]
Comment (benbellick) [reply packages/ppx_deriving_decoders/ppx_deriving_decoders.0.1/opam:16]: Ah addressing this one, but amazing! Thanks for the help 😄

**Comment by shonfeder** [comment]
Great! The lower bounds checks are passing and only the decoder issue persists, and we’ll tackle that up stream. Thanks very much for authoring and sharing this useful package :)

Consider announcing it on discuss.ocaml.org :)

**Pull request merged**